fabric-recipe-api-v1/src/main/java/net/fabricmc/fabric/api/recipe/v1/VanillaRecipeBuilders.java

```diff
@@ -0,0 +1,395 @@
+	 */
+	public static SmeltingRecipe smeltingRecipe(Identifier id, String group, Ingredient input, ItemStack output, float experience, int cookTime) {
+		if (input == Ingredient.EMPTY) throw new IllegalArgumentException("Input cannot be empty.");
+		if (cookTime < 0) throw new IllegalArgumentException("Cook time must be equal or greater than 0");
+
+		return new SmeltingRecipe(id, group, input, output, experience, cookTime);
+	}
+
+	/**
+	 * Returns a new blasting recipe.
+	 *
+	 * @param id         the identifier of the recipe
+	 * @param group      the group of the recipe
+	 * @param input      the input ingredient
+	 * @param output     the output item stack
+	 * @param experience the experience given
+	 * @param cookTime   the cook time in ticks
+	 * @return the blasting recipe
+	 */
+	public static BlastingRecipe blastingRecipe(Identifier id, String group, Ingredient input, ItemStack output, float experience, int cookTime) {
+		if (input == Ingredient.EMPTY) throw new IllegalArgumentException("Input cannot be empty.");
+		if (cookTime < 0) throw new IllegalArgumentException("Cook time must be equal or greater than 0");
+
+		return new BlastingRecipe(id, group, input, output, experience, cookTime);
+	}
+
+	/**
+	 * Returns a new smoking recipe.
+	 *
+	 * @param id         the identifier of the recipe
+	 * @param group      the group of the recipe
+	 * @param input      the input ingredient
+	 * @param output     the output item stack
+	 * @param experience the experience given
+	 * @param cookTime   the cook time in ticks
+	 * @return the smoking recipe
+	 */
+	public static SmokingRecipe smokingRecipe(Identifier id, String group, Ingredient input, ItemStack output, float experience, int cookTime) {
+		if (input == Ingredient.EMPTY) throw new IllegalArgumentException("Input cannot be empty.");
+		if (cookTime < 0) throw new IllegalArgumentException("Cook time must be equal or greater than 0");
+
+		return new SmokingRecipe(id, group, input, output, experience, cookTime);
+	}
+
+	/**
+	 * Returns a new campfire cooking recipe.
+	 *
+	 * @param id         the identifier of the recipe
+	 * @param group      the group of the recipe
+	 * @param input      the input ingredient
+	 * @param output     the output item stack
+	 * @param experience the experience given
+	 * @param cookTime   the cook time in ticks
+	 * @return the campfire cooking recipe
+	 */
+	public static CampfireCookingRecipe campfireCookingRecipe(Identifier id, String group, Ingredient input,
+															ItemStack output, float experience, int cookTime) {
+		if (input == Ingredient.EMPTY) throw new IllegalArgumentException("Input cannot be empty.");
+		if (cookTime < 0) throw new IllegalArgumentException("Cook time must be equal or greater than 0");
+
+		return new CampfireCookingRecipe(id, group, input, output, experience, cookTime);
+	}
+
+	/**
+	 * Represents a shaped crafting recipe builder.
+	 */
+	public static final class ShapedRecipeBuilder {
```

Comment:
I would split this class up a bit, I dont see any benefit to having all this in one place.